src/coreComponents/physicsSolvers/fluidFlow/ReactiveCompositionalMultiphaseOBL.cpp

```diff
@@ -339,7 +354,7 @@ real64 ReactiveCompositionalMultiphaseOBL::calculateResidualNorm( DomainPartitio
 
   if( getLogLevel() >= 1 && logger::internal::rank == 0 )
   {
-    std::cout << GEOSX_FMT( "    ( Rfluid ) = ( {:4.2e} ) ;", residual );
+    std::cout << GEOSX_FMT( "    ( Rflow ) = ( {:4.2e} ) ;", residual );
   }
```

Comment:
Couldn't this be replaced with `GEOSX_LOG_LEVEL_RANK_0`?


definitions:
```
#define GEOSX_LOG_LEVEL_RANK_0( minLevel, msg ) GEOSX_LOG_RANK_0_IF( this->getLogLevel() >= minLevel, msg )

#define GEOSX_LOG_RANK_0_IF( EXP, msg ) \
  do { \
    if( ::geosx::logger::internal::rank == 0 && EXP ) \
    { \
      std::ostringstream oss; \
      oss << msg; \
      std::cout << oss.str() << std::endl; \
    } \
  } while( false )

```

Reply:
Yes, sounds good, I made the modification.

I think we have used `if( getLogLevel() >= 1 && logger::internal::rank == 0 )` instead of `GEOSX_LOG_LEVEL_RANK_0` in all single-physics solvers to have a concise output in our simulation logs (taking the example of a poromechanics simulation with wells):
```
Attempt:  0, ConfigurationIter:  0, NewtonIter:  3                                                                                                                          
( Rsolid ) = ( 1.41e-13 ) ;     ( Rflow ) = ( 5.45e-03 ) ;     ( Rwell ) = ( 3.27e-07 ) ;     ( R ) = ( 5.45e-03 ) ;  
```
instead of:
```
Attempt:  0, ConfigurationIter:  0, NewtonIter:  3                                                                                                                          
( Rsolid ) = ( 1.41e-13 ) ;
( Rflow ) = ( 5.45e-03 ) ;
( Rwell ) = ( 3.27e-07 ) ;
( R ) = ( 5.45e-03 ) ;  
```
where the line breaks come from the `std::endl` in GEOSX_LOG_LEVEL_RANK_0`. 

But maybe we should just let the coupled solver issue the residual norm values in a uniform way [here](https://github.com/GEOSX/GEOSX/blob/cd04f12c24a0004432673ce5c667512175c95ffd/src/coreComponents/physicsSolvers/multiphysics/CoupledSolver.hpp#L245) and remove all this norm logging from the single-physics solvers. This would make it easier to work on a more concise/organized formatting (maybe with a table or something like that).

Reply:
I may or may not have said this before, but in my opinion the appropriate place to do residual norm logging is in `SolverBase::solveNonlinearSystem`, in the middle of Newton loop, not the calculation functions. If we need to modify the signature of `calculateResidualNorm` to return multiple values (a vector of sub-norms), let's do it. Otherwise we get into a nesting problem: what if a sub-solver of a coupled solver is itself a coupled solver? Only the solver that is executing the nonlinear loop has the correct information about what/when/how to print (are we in a middle of a line search? what level of detail has the user requested?). Also single responsibility principle suggests that a compute function should not also do logging.